### PR TITLE
feat(scrum-86): supabase auth integration, auth UI

### PR DESCRIPTION
**Pull request by emachta**
added create account flow, validation, integration with supabase login/create account, auth context provider, and supabase OAuth providers

SIGN UP FLOW:

https://github.com/user-attachments/assets/8b7a8f51-cc39-474c-b167-54ea9e82e7ef

LOGIN FLOW:

https://github.com/user-attachments/assets/23bbafe0-5954-455a-8455-1a809148a679

Supabase Auth dashboard:
Screenshot 2024-11-27 at 11 26 01 PM

### CODE REVIEW

File: App.tsx [hunk 1]
```diff
@@ -1,15 +1,19 @@
...
 import tw from './tailwind';
 import BottomTabNavigation from './src/Navigation/BottomStackNavigation';
-import supabase from '@supabase/supabase-js';
-import {SafeAreaProvider} from 'react-native-safe-area-context';
+import {useAuth} from './src/Hooks/Auth';
+import {AuthContext} from './src/Context/AuthContext';
 
 function App(): React.JSX.Element {
+  // grab current auth details and pass to context for global access
+  const session = useAuth();
```
Comment (emachta) [inline App.tsx:10]: adds auth hook, and context provider to allow global access to auth anywhere in app

File: package.json [hunk 2]
```diff
@@ -10,6 +10,7 @@
     "test": "jest"
   },
   "dependencies": {
+    "@react-native-async-storage/async-storage": "^2.1.0",
```
Comment (emachta) [inline package.json:13]: required for supabase auth

File: src/Components/Inputs/LoginInput.tsx [hunk 3]
```diff
@@ -0,0 +1,66 @@
+import {View, Text, ImageBackground, Image, TextInput} from 'react-native';
```
Comment (emachta) [inline src/Components/Inputs/LoginInput.tsx:1]: input field for login, moved into its own component

File: src/Screens/Profile/SignUp.tsx [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+        />
+        <LoginInput
+          autoComplete="new-password"
+          setText={text => setFormState({...formState, password: text})}
+          text={formState.password}
+          errorState={errorState.password}
+          defaultText="Password"
+          secureTextEntry
+          keyboardType="default"
+        />
```
Comment (ojandali93) [inline src/Screens/Profile/SignUp.tsx:96]: Hey can you add just a few things. With the password and confirm password, can you add some way for the user to know that a password needs to be a minimum number of charactors, one capital, and one number if that is how the password is setup. some type of validation to make sure the user created a valid password.

File: src/Screens/Profile/SignUp.tsx [hunk 5]
```diff
@@ -0,0 +1,160 @@
+import {View, Text, Image, TouchableOpacity, Alert} from 'react-native';
```
Comment (ojandali93) [inline src/Screens/Profile/SignUp.tsx:1]: In general for the profile creation screen, if the input is not valid, show a red border but if it is a valid input, show a green border? This is something we can do later, just wanted to let you know.

**Comment by ojandali93** [comment]
I left comments in the code changes if you look at the code. 

they are the following, 

if we can add some type of validation for something like password to show the user that a password has to have a certain amount of chars, one capital, one number, and special char if we want to require that. Also validation that shows the passwords match. Also make sure that the email contains an @ and a (.). 

The other thing would be something like if the inputs pass validation, then make the border green. 

This can be another feature that is a new feature that builds on the current feature that is merged. I have no issue with that.

**Review by ojandali93 — APPROVED** [review]

**Pull request merged**